src/app/core/views/Preferences/tabs/Account/AccountDetails.tsx

```diff
@@ -161,37 +176,165 @@ function AccountDetailsModal({
+    } else {
+      try {
+        setStatus({ tag: 'loading' });
+        const correctPassword = await areCredentialsCorrect(email, password);
+        if (correctPassword) {
+          // TODO -> Send verificaion email
+          // Send verification to newEmail
+          notificationsService.show({
+            text: translate('views.account.tabs.account.accountDetails.changeEmail.sucessSendingVerification', {
+              email: newEmail,
+            }),
+            type: ToastType.Success,
+          });
+          onClose();
+        } else {
+          setStatus({ tag: 'error', type: 'PASSWORD_INVALID' });
+        }
+      } catch {
+        setStatus({ tag: 'error', type: 'UNKNOWN' });
+        notificationsService.show({
+          text: translate('views.account.tabs.account.accountDetails.changeEmail.errorSendingVerification'),
+          type: ToastType.Error,
+        });
+      }
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <form className="flex flex-col space-y-5" onSubmit={onChange}>
+        <h1 className="text-2xl font-medium text-gray-80">
+          {translate('views.account.tabs.account.accountDetails.changeEmail.title')}
+        </h1>
+
+        <div className="flex flex-col space-y-3">
+          <Input
+            disabled
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.email') as string}
+            value={email}
+            name="email"
+          />
+          <Input
+            required
+            variant="email"
+            autoComplete="off"
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.newEmail') as string}
+            onChange={setNewEmail}
+            accent={status.tag === 'error' && status.type === 'SAME_EMAIL' ? 'error' : undefined}
+            message={
+              status.tag === 'error' && status.type === 'SAME_EMAIL'
+                ? (translate('views.account.tabs.account.accountDetails.changeEmail.errorSameEmail') as string)
+                : undefined
+            }
+            name="newemail"
+          />
+          <Input
+            required
+            disabled={status.tag === 'loading'}
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.password') as string}
+            variant="password"
+            onChange={setPassword}
+            accent={status.tag === 'error' && status.type === 'PASSWORD_INVALID' ? 'error' : undefined}
+            message={
+              status.tag === 'error' && status.type === 'PASSWORD_INVALID'
+                ? (translate('views.account.tabs.account.accountDetails.changeEmail.errorPassword') as string)
+                : undefined
+            }
+            name="password"
+          />
+        </div>
+
+        <div className="flex justify-end space-x-2">
+          <Button disabled={status.tag === 'loading'} variant="secondary" onClick={onClose}>
```

Comment:
Here also, it is repeated more than twice. 
`const isLoadingStatus = status.tag === 'loading';`